packages/mds-utils/utils.ts

```diff
@@ -770,6 +772,44 @@ function normalizeToArray<T>(elementToNormalize: T | T[] | undefined): T[] {
   return [elementToNormalize]
 }
 
+export interface ProviderClaimResponse extends ApiResponse {
+  locals: ApiResponseLocals & {
+    provider_id: UUID
+  }
+}
+
+async function providerClaimMiddleware(req: ApiRequest, res: ProviderClaimResponse) {
+  if (res.locals.claims === null) {
+    res.status(400).send({
+      result: 'No claims provided'
+    })
+    return false
+  }
+
+  const { provider_id } = res.locals.claims
+
+  if (!isUUID(provider_id)) {
+    await log.warn(req.originalUrl, req.method, 'invalid provider_id is not a UUID', provider_id)
+    res.status(400).send({
+      result: `invalid provider_id ${provider_id} is not a UUID`
+    })
+    return false
+  }
+
+  if (!isProviderId(provider_id)) {
+    await log.warn(req.originalUrl, req.method, 'invalid provider_id is not a known provider', provider_id)
+    res.status(400).send({
+      result: `invalid provider_id ${provider_id} is not a known provider`
+    })
+    return false
+  }
+
+  res.locals.provider_id = provider_id
+
+  log.info(providerName(provider_id), req.method, req.originalUrl)
+  return provider_id
```

Comment:
Use `return next()`. Routes using this middleware should get the `provider_id` from `res.locals.provider_id`